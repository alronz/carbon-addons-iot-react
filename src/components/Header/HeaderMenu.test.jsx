import React from 'react';
import { mount } from 'enzyme';
import { HeaderMenuItem } from 'carbon-components-react/lib/components/UIShell';

import HeaderMenu from './HeaderMenu';

describe('HeaderMenu', () => {
  let mountNode;
  let mockProps;

  beforeEach(() => {
    mountNode = document.createElement('div');
    mockProps = {
      'aria-label': 'Accessibility label',
      className: 'custom-class',
      // We use `ref` instead of `focusRef` becase `HeaderMenu` forwards the ref
      // to the underlying menu button
      ref: jest.fn(),
      tabIndex: -1,
    };

    document.body.appendChild(mountNode);
  });

  afterEach(() => {
    mountNode.parentNode.removeChild(mountNode);
  });

  const wrapper = mount(
    <HeaderMenu {...mockProps}>
      <HeaderMenuItem href="/a">A</HeaderMenuItem>
      <HeaderMenuItem href="/b">B</HeaderMenuItem>
      <HeaderMenuItem href="/c">C</HeaderMenuItem>
    </HeaderMenu>,
    {
      attachTo: mountNode,
    }
  );
  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render aria-label', () => {
    const wrapper = mount(
      <HeaderMenu {...mockProps}>
        <HeaderMenuItem href="/a">A</HeaderMenuItem>
        <HeaderMenuItem href="/b">B</HeaderMenuItem>
        <HeaderMenuItem href="/c">C</HeaderMenuItem>
      </HeaderMenu>,
      {
        attachTo: mountNode,
      }
    );
    const headerMenu = wrapper.childAt(0);
    const headerMenuText = headerMenu.find('.bx--header__menu-title').text();

    expect(headerMenuText).toMatch('Accessibility label');
  });

  it('should render content prop', () => {
    const menuContent = () => <p>Some other text</p>;
    const wrapper = mount(
      <HeaderMenu renderMenuContent={menuContent} {...mockProps}>
        <HeaderMenuItem href="/a">A</HeaderMenuItem>
        <HeaderMenuItem href="/b">B</HeaderMenuItem>
        <HeaderMenuItem href="/c">C</HeaderMenuItem>
      </HeaderMenu>,
      {
        attachTo: mountNode,
      }
    );

    const headerMenu = wrapper.childAt(0);
    const headerMenuAnchorChildText = headerMenu
      .find('.bx--header__menu-title')
      .childAt(0)
      .text();
    const headerMenuText = headerMenu.find('.bx--header__menu-title').text();

    expect(headerMenuText).not.toMatch('Accessibility label');
    expect(headerMenuAnchorChildText).toMatch('Some other text');
  });
});
