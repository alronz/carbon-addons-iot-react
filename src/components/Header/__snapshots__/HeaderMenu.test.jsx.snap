// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu should render 1`] = `
<ForwardRef>
  <HeaderMenu
    focusRef={null}
    isMenu={true}
    renderMenuContent={[Function]}
  >
    <li
      className="bx--header__submenu"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
    >
       
      <a
        aria-expanded={false}
        aria-haspopup="menu"
        className="bx--header__menu-item bx--header__menu-title"
        href="javascript:void(0)"
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <defaultRenderMenuContent>
          <ChevronDownGlyph
            className="bx--header__menu-arrow"
            focusable={false}
            height={7}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 12 7"
            width={12}
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              aria-hidden={true}
              className="bx--header__menu-arrow"
              focusable={false}
              height={7}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 12 7"
              width={12}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.002 5.55L11.27 0l.726.685L6.002 7 0 .685.726 0z"
              />
            </svg>
          </ChevronDownGlyph>
        </defaultRenderMenuContent>
      </a>
      <ul
        className="bx--header__menu"
        role="menu"
      >
        <ForwardRef
          href="/a"
          key=".0"
          role="none"
        >
          <li
            role="none"
          >
            <Link
              className="bx--header__menu-item"
              element="a"
              href="/a"
              role="menuitem"
              tabIndex={0}
            >
              <a
                className="bx--header__menu-item"
                href="/a"
                role="menuitem"
                tabIndex={0}
              >
                <span
                  className="bx--text-truncate--end"
                >
                  A
                </span>
              </a>
            </Link>
          </li>
        </ForwardRef>
        <ForwardRef
          href="/b"
          key=".1"
          role="none"
        >
          <li
            role="none"
          >
            <Link
              className="bx--header__menu-item"
              element="a"
              href="/b"
              role="menuitem"
              tabIndex={0}
            >
              <a
                className="bx--header__menu-item"
                href="/b"
                role="menuitem"
                tabIndex={0}
              >
                <span
                  className="bx--text-truncate--end"
                >
                  B
                </span>
              </a>
            </Link>
          </li>
        </ForwardRef>
        <ForwardRef
          href="/c"
          key=".2"
          role="none"
        >
          <li
            role="none"
          >
            <Link
              className="bx--header__menu-item"
              element="a"
              href="/c"
              role="menuitem"
              tabIndex={0}
            >
              <a
                className="bx--header__menu-item"
                href="/c"
                role="menuitem"
                tabIndex={0}
              >
                <span
                  className="bx--text-truncate--end"
                >
                  C
                </span>
              </a>
            </Link>
          </li>
        </ForwardRef>
      </ul>
       
    </li>
  </HeaderMenu>
</ForwardRef>
`;
